omod/src/main/webapp/resources/htmlforms/003-HMIS_HIVCare_ARTCard-ClinicalAssessmentPage.xml

```diff
@@ -0,0 +1,1531 @@
+                if (encounterDate &lt; date_viral_load_taken) {
+                    getField('date-viral-load-taken.error').html('The date the viral load sample was taken cannot be after the date of this visit ' + encounterDate).show();
+                    return false;
+                }
+            }
+
+            /* If the viral load is detected, the number of copies must be greater than 20 which is the lowest detectable level for a machine */
+            if ((viral_load_qualitative == 1301) &amp;&amp; (getValue('viral-load-results-copies.value') &lt; 20)) {
+                getField('viral-load-results-copies.error').html('Please enter the number of virus copies per ml. the minimum number of copies is 20').show();
+                return false;
+            }
+
+            /* if DSDM is not FBIM or FTR or Empty and patient is not stable should not save.*/
+            if (jq("#patient-categorization").find("select").find(":selected").val() !== "90003" &amp;&amp; (jq("#165143").find("select").find(":selected").val() !== "165138" &amp;&amp; jq("#165143").find("select").find(":selected").val() !== "165140" &amp;&amp; jq("#165143").find("select").find(":selected").val() !== "")) {
+                getField('165143.error').html('Patient Not Stable due to ' + instabilityCause + '. \n Only Stable Patients can be enrolled in ' + jq("#165143").find("select").find(":selected").text()).show();
+
+                jq().toastmessage('showErrorToast', " Patient Not Stable " + instabilityCause + ". \n Only Stable Patients can be enrolled in " + jq("#165143").find("select").find(":selected").text());
+                return false;
+            }
+            return true;
+        });
+    }
+
+        <!-- only show BMI if patient is more than 5 years old (on the encounterDate) -->
+        <includeIf velocityTest="$patient.getAge($encounter.getEncounterDatetime()) > 5">
+
+            var calculateBmi = function(wt, ht) {
+            var bmi = null
+            if (wt &amp;&amp; ht) {
+            bmi = wt / ((ht/100) * (ht/100));
+            }
+            return bmi ;
+            }
+
+            <ifMode mode="ENTER">
+
+                //functions to handle updating the bmi when in ENTER mode
+                    var updateBmi = function() {
+                    var wt = htmlForm.getValueIfLegal('weight.value');
+                    var ht = htmlForm.getValueIfLegal('height.value');
+
+                    var bmi = calculateBmi(wt, ht);
+
+                    if (bmi == null || isNaN(bmi)) {
+                        jq('#calculated-bmi-wrapper').hide();
+                    } else {
+                        jq('#no-calculated-bmi').hide();
+                        jq('#calculated-bmi-wrapper').show();
+                        jq('#calculated-bmi').html(bmi.toFixed(1));
+                        jq('#hidden-calculated-bmi').val(bmi.toFixed(1));
+                    }
+                };
+
+                jq(function() {
```

Comment:
@semujju why do you have to put all these functions in separate ja(function(){}); tags? they should all be in one because it makes no difference

Reply:
do